sdk/nodejs/apigateway/metrics.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as cloudwatch from "../cloudwatch";
+
+export type ApigatewayMetricName =
+    "4XXError" | "5XXError" | "CacheHitCount" | "CacheMissCount" |
+    "Count" | "IntegrationLatency" | "Latency";
+
+/**
+ * Creates an AWS/ApiGateway metric with the requested [metricName]. See
+ * https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-metrics-and-dimensions.html
+ * for list of all metric-names.
+ *
+ * Note, individual metrics can easily be obtained without supplying the name using the other
+ * [metricXXX] functions.
+ *
+ * You can use the dimensions in the following table to filter API Gateway metrics.
+ *
+ * 1. "ApiName": Filters API Gateway metrics for an API of the specified API name.
+ * 2. "ApiName, Method, Resource, Stage": Filters API Gateway metrics for an API method of the
+ *    specified API, stage, resource, and method.
+ *
+ *    API Gateway will not send such metrics unless you have explicitly enabled detailed CloudWatch
+ *    metrics. You can do this in the console by selecting Enable CloudWatch Metrics under a stage
+ *    Settings tab. Alternatively, you can call the stage:update action of the API Gateway REST API
+ *    to update the metricsEnabled property to true.
+ *
+ *    Enabling such metrics will incur additional charges to your account. For pricing information,
+ *    see Amazon CloudWatch Pricing.
+ * 3. "ApiName, Stage": Filters API Gateway metrics for an API stage of the specified API and stage.
+ */
+export function metric(metricName: ApigatewayMetricName, change: cloudwatch.MetricChange = {}) {
+    return new cloudwatch.Metric({
+        namespace: "AWS/ApiGateway",
+        name: metricName,
+        ...change,
+    });

```

Comment:
always have teh general 'metric' helper that all specific metricXXX helpers defer to. This has the most docs on how the metrics for this package work (some packages go very much in depth). We should likely have these become README.md info for that package.